zerver/lib/subscription_info.py

```diff
@@ -374,6 +363,7 @@ def validate_user_access_to_subscribers_helper(
     stream_dict: Mapping[str, Any],
     check_user_subscribed: Callable[[UserProfile], bool],
     user_recursive_group_ids: set[int],
+    should_fetch_user_recursive_group_ids: bool = False,
```

Comment:
I think a better interface for this would be to make `user_recursive_group_ids` accept a `set[int] | None`, and in the `None` case, fetch it; then we would not need two parameters.